src/lgar.cxx

```diff
@@ -347,9 +347,9 @@ extern void InitFromConfigFile(string config_file, struct model_state *state)
 
       continue;
     }
-    else if (param_key == "max_soil_types") {
-      state->lgar_bmi_params.num_soil_types = stod(param_value);
-      is_max_soil_types_set = true;
+    else if (param_key == "max_valid_soil_types") {
+      state->lgar_bmi_params.num_soil_types = fmax(stod(param_value), 25.0);
```

Comment:
why is num_soil_types a floating point value?

Reply:
also, why use 25 here?  On line 537 the default maximum number is 12.

Reply:
fixed! removed 25 and using `stoi`